src/Chapter1.hs

```diff
@@ -416,7 +416,7 @@ task is to specify the type of this function.
 >>> squareSum 3 4
 49
 -}
-
+squareSum :: Num a => a -> a -> a
```

Comment:
We explain `Num` in the following chapters, so you are one step ahead 😸

In here `Int -> Int -> Int` would also work, as a more specific type 🙂

Reply:
Thanks for your inputs !

Would 
```hs
squareSum :: Int a => a -> a -> a
```

be equivalent or it means something else ?

Reply:
The the following signature is invalid:

```haskell
squareSum :: Int a => a -> a -> a
```

`Num` is one of standard typeclasses (aka interfaces), we explain them in Chapter Tree. And `Int` is not a typeclass, so you can't specify `Int a` in the type signature, it will be a compiler error.

On a more advanced note, the following type signature is equivalent to `Int -> Int -> Int`:

```haskell
squareSum :: Int ~ a => a -> a -> a
```

Here we use `~` which is a type equality. So you can use the `a` type variable and it will be an alias to `Int` on simple words. But we don't cover this course, such tricks are needed only in very advanced cases or when trying to achieve something non-trivial, no need to worry about that for now :slightly_smiling_face: